src/backend/utils/adt/json.c

```diff
@@ -1374,3 +1374,28 @@ json_typeof(PG_FUNCTION_ARGS)
 
 	PG_RETURN_TEXT_P(cstring_to_text(type));
 }
+
+
+/*
+ * SQL function tsql_json_build_object()
+ * 
+ * Transform the input key,val into json key:val pair
+ */
+StringInfo
+tsql_json_build_object(Datum colname, Datum colval, Oid collation, bool is_null)
+{
+	StringInfo	result;
+
+	result = makeStringInfo();
+	
+	add_json(colname, false, result, CSTRINGOID, true);
+
+	appendStringInfoString(result, ":");
+	if (is_null)
+		add_json(colval, true, result, collation, false);
+	else
+		add_json(colval, false, result, collation, false);
+	
+	return result;
```

Comment:
Returning a pointer declared in a function is risky. Let's pass `StringInfo` from the caller as an input argument.